### PR TITLE
Make the Blocks navigation a nested list to better communicate the blocks nesting level

### LINKED ISSUE #11713: Blocks Navigation menu: no semantic information for nested blocks
When the Blocks Navigation menu displays nested blocks, it visually uses some indentation:

blocks navigation menu nesting 10545

While the visual indication is clear, there's no semantic information or any other kind of information assistive technologies can use to communicate the blocks nesting level. Blocks are announced as if they were in a flat list.

The nesting level is an important information that needs to be communicated to all users. I'm not sure the Menu component is suitable for this purpose. While [ARIA menus can have sub-menus](https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-1/menubar-1.html), I;d argue the information this component conveys is closer to an ARIA tree.

On the other hand, an ARIA tree is complex to interact with. Considering the menu items should be links as proposed in #11711 I'd recommend to keep things simple and just use a navigation menu that has the ability to render nested unordered lists when necessary. The item "checked" state could be communicated simply using some screen reader text. Nested lists give us the information we need for free:

Simple example: https://www.w3.org/WAI/tutorials/page-structure/content/#nested-lists

screenshot 134

screenshot 2018-11-10 at 14 17 20

### PR DESCRIPTION
**Pull request by afercia**
## Description

When there are nested blocks, the Blocks navigation displays them with some indentation. However, this information is only visual. This PR seeks to communicate the nesting level also semantically by using an unordered list. Lists convey important information for free: the number of items in the list, their nesting level, the current item number.

Assistive technologies can use the list semantics to properly inform users, so they can actually get the structure of nested blocks. 

I've done my best to avoid visual changes.

Worth noting arrows navigation won't work well with screen readers on Windows because now there isn't an ARIA role to instruct screen readers to stop intercepting keystrokes. I think providing the missing information is more important than arrows navigation. Also, there are proposal to change and extend the functionalities of the Blocks navigation, see #10984, #11408, #11688, and #11711. In all those cases, an ARIA menu wouldn't be the proper UI to use and arrows navigation would need to be removed anyways.

I'm trying to keep the arrows navigation behavior for non-screen reader users for now, but I'd tend to think this component is now more similar to the "Document outline": a simple list with no special interaction other than the native one. Please do let me know if completely removing arrows navigation is desirable.

- makes the Blocks Navigation an unordered (nested) list 
- uses `NavigableMenu` to keep arrows navigation (for now)
- uses a `screen-reader-text` to communicate the selected state of a list item 
- avoids to render an `aria-orientation` attribute, as this is not a menu any longer
- changes the `editor-block-icon` to a `<span>` element: `<div>` elements are invalid HTML within buttons; we'd want this component to be reusable everywhere and also ensure valid HTML and a div doesn't seem appropriate; the `editor-block-icon` is already a flex element with set width and height so this change shouldn't harm anything
- "Block Navigation" is not used as target for `aria-labelledby` any longer, I'm not so worried about it because users get this popover clicking on a button that already announces "Block Navigation"

Fixes #11713 

Screenshots to illustrate how the nesting level gets announced thanks to the native semantics of an `<ul>` element:

screenshot 2018-11-11 at 12 29 19

screenshot 2018-11-11 at 12 29 27

screenshot 2018-11-11 at 12 28 17

Note: other screen readers communicate the nesting info in different ways, see the example screenshots on #11713

### CODE REVIEW

**Comment by jwold** [comment]
From a UI perspective this looks great. Seems like you've been able to propose a solution that still keeps the look and feel we've been using.

**Review by tofumatt — APPROVED** [review]
This is awesome! UI still looks great and the accessibility improvements are ace. Thanks a bunch, this is great 👍

File: packages/editor/src/components/block-navigation/index.js [hunk 1]
```diff
@@ -25,23 +25,30 @@ function BlockNavigationList( {
 	showNestedBlocks,
 } ) {
 	return (
-		<ul className="editor-block-navigation__list" role="presentation">
+		/*
+		 * Disable reason: The `list` ARIA role is redundant but
```
[the remark this answers is not in the document]
Comment (afercia) [reply packages/editor/src/components/block-navigation/index.js]: Not sure. It's a Safari specific bug. Chrome is exempt. I wouldn't know where to look for Apple open bugs. We do the same thing in `BlockTypesList`, for more details see https://github.com/WordPress/gutenberg/pull/7058#issue-191802381. Test case: https://codepen.io/afercia/full/WxmJWx/

File: packages/editor/src/components/block-navigation/index.js [hunk 2]
```diff
@@ -55,6 +62,7 @@ function BlockNavigationList( {
 				);
 			} ) }
 		</ul>
+		/* eslint-enable jsx-a11y/no-redundant-roles */
```
[the remark this answers is not in the document]
Comment (afercia) [reply packages/editor/src/components/block-navigation/index.js]: Tried and I'd need to do something like this to make the linter happy:
```
		/* eslint-disable jsx-a11y/no-redundant-roles */
		<ul className="editor-block-navigation__list" role="list">
			{ /* eslint-enable jsx-a11y/no-redundant-roles */ }
			{ map( blocks, ( block ) => {
```
not sure it looks so nice 🙂

**Pull request merged**